### ISSUE TITLE
issue running recent polkadot updates in React Native (no issue in node)

### ISSUE THREAD

**Issue by gre** [polkadot-js/api#3953]
We have recently attempted to upgrade from lib 5.5.1 to 5.8.3 and this seems to break in React Native but not in Node.js.

```
undefined is not an object (evaluating 's.type.asNMap.value'
```

during the phase of scanning accounts (with Ledger Live logic).

I can't dive more precisely yet where the problem is, but definitely related to that bump. Maybe someone have a better idea?

There are no breaking changes announced between these versions so I would have expected no breaking changes for this upgrade path.

**Comment by jacogr** [comment]
Indeed. This is weird. The only map-related changes I can think of was allowing queries by all keys, but that was also no -breaking. 

Do you have the problematic source on your side, that may yield an idea.

**Comment by jacogr** [comment]
I would refer that metadata fetch to sidecar, possibly something is amis in the result as returned - the API handles normal metadata via RPC nodes correctly. (As evident in tools where the metadata is fetched from the chain)

As an aside, I am really not sure where `this.options.fatal` that is referred to above comes from. Searching the API repo source yields no results for that occurance. (In addition to locally searching the common repo, no results found - unless my VSCode is broken atm... which has hapened in the past)

**Comment by TarikGul** [comment]
> I would refer that metadata fetch to sidecar, possibly something is amis in the result as returned

To give some info/insight on this, the response is fairly simple under the hood. It's only using `api.rpc.state.getMetadata(<hash?>)` and it doesnt tamper with the data, it just applies `.toHex()` to it. I went back and double checked each metadata response from sidecar against raw polkadot-js calls and they all matched.  Checked both @polkadot/api 6.3.1 and 6.4.1.

**Comment by jacogr** [comment]
@TarikGul Yeap, I pulled down the sidecar source and looked at it - that is why I added the tests in #4077 to explicitly verify that what comes in indeed comes out.

**Comment by jacogr** [comment]
I looked at that `TextDecoder` polyfill used  - it accepts 2 arguments, the charset and then options. So options are undefined and thus `options.fatal` would throw. This is certainly related to that library, there is actually no options in the base JS `TextDecoder`. See -

https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder

Where `TextDecoder` is used in `@polkadot/util`, it passes a single argument, when not specified it should use utf-8 anyway (as per the above documentation), so passing it is actually really not needed - 

https://github.com/polkadot-js/common/blob/master/packages/util/src/u8a/toString.ts#L6

As an aside, if there is no class present in the environment, the `@polkadot` libs will provide one, hence the use of `x-textdecoder` internally. In this case since there is a polyfill injected, so it uses that.

**Comment by jacogr** [comment]
On which chain? The “seq-uence” is not a type I have see before anywhere in metadata. There is “sequence” (no -) in the metadata type definitions.
